src/common/router.js

```diff
@@ -29,6 +34,7 @@ export function setRoute(hash, replace) {
   if (replace) {
     window.history.replaceState('', null, hashString);
   } else {
+    route.stack.push(Object.assign({}, route));
```

Comment:
How about just picking `pathname` and `query` here?
```js
route.stack.push({
  pathname: route.pathname,
  query: route.query,
});
```
So we don't have to define `stack` in a different way.

Reply:
You didn't include `paths` and what if in the future you add another property and it needs to be explicitly copied here? Nah. Let's see if you like the new code better.

Reply:
`paths` is for convenience and thus not necessary to be cached, only `pathname` and `query` are real data. It can even be removed I think.

Reply:
So what about the new code?

Reply:
How about exporting last route as a function?

Reply:
Okay.